## Idea: New bus has a shorter run time compare to the old X8 bus
### Null hypothesis:
#### new bus run time is longer than or the same as the old bus run time
$H_0$ : Time_new  >=  Time_old
  

### Alternative hypothesis:
#### new bus run time is shorter than the old bus run time
$H_0$ : Time_new  < Time_old

### Significant level   $\alpha=0.05$
#### In this case, we need to compare the absolute value of Z-score that we get from our data with 1.64 to determine our result

In [22]:
from __future__ import print_function
__author__= 'yc3420'

import pandas as pd
import os

%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [23]:
!curl https://raw.githubusercontent.com/fedhere/PUI2018_fb55/master/Lab4_fb55/times.txt > time.csv

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1179  100  1179    0     0   7076      0 --:--:-- --:--:-- --:--:--  7102


In [24]:
print("PUIdata", os.getenv("PUIDATA"))

PUIdata /nfshome/yc3420/PUIdata


In [5]:
!mv time.csv $PUIDATA

In [8]:
!ls $PUIDATA

201503-citibike-tripdata.csv  drinking_water_quaity.csv
201503-citibike-tripdata.zip  time.csv


In [13]:
bus_time = pd.read_csv(os.getenv("PUIDATA") + '/' + 'time.csv', names = ['bus_dur'])

In [14]:
bus_time.head()

bus_dur
0  31.622239
1  32.821376
2  30.229101
3  31.413766
4  39.010550

In [31]:
mean_o = 36.0
std_o = 6.0
N = len(bus_time)
#print (N)
mean_n = bus_time['bus_dur'].mean().astype(float)
std_n = bus_time['bus_dur'].std().astype(float)
#print (mean_n,std_n)
z_score = (mean_o - mean_n) / (std_o/N**0.5)
print ("z score = ", z_score)
print ("z score abs = ", abs(z_score))

z score =  2.55639718617
z score abs =  2.55639718617


### Here the z score we get is 2.556 which is greater than 1.645 (α = 0.05 , one tail). So we could reject the H0 hypothesis at the 95% confidence level. 

## conclusion: the new bus has a shorter duration compare to the old one